ct_core/include/ct/core/systems/continuous_time/linear/LTISystem.h

```diff
@@ -130,6 +130,34 @@ class LTISystem : public LinearSystem<STATE_DIM, CONTROL_DIM>
         }
     }
 
+    //! computes the controllability gramian by discretization
+    /*!
+     * Computes the controllability gramian
+     *
+     * @return Returns the controllability Gramian
+     */
+    Eigen::Matrix<double, STATE_DIM, STATE_DIM>& computeControllabilityGramian(double tolerance=1e-9)
+    {
+        Eigen::Matrix<double, STATE_DIM, STATE_DIM> CG_prev = Eigen::Matrix<double, STATE_DIM, STATE_DIM>::Zero();
+        Eigen::Matrix<double, STATE_DIM, STATE_DIM> CG      = Eigen::Matrix<double, STATE_DIM, STATE_DIM>::Zero();
+        Eigen::Matrix<double, STATE_DIM, STATE_DIM> A_prev = Eigen::Matrix<double, STATE_DIM, STATE_DIM>::Identity();
+        while (true)
+        {
+            CG = CG_prev + A_prev * B_ * B_.transpose() * A_prev.transpose();
```

Comment:
In my understanding, this equation holds for the discrete-time case, does it not?
This class, however, represents the continous-time LTI system. So if I am not mistaken, these methods should go to DiscreteLinearSystem.h.

Reply:
Had to move this over to discretelinearsystem... will update this.